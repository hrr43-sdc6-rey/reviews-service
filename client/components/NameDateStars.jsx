import React from 'react';
import PropTypes from 'prop-types';
import ReviewStars from './ReviewStars.jsx';

function NameDateStars({
  username, month, year, stars,
}) {
  return (
    <div>
      <div className="nameDate">
        <p className="username">
          {username}
        </p>
        <div className="date">
          <div className="dot">&#183;</div>
          <div className="month">
            {month}
          </div>
          {year}
        </div>
      </div>
      <div className="reviewStars">
        <ReviewStars stars={stars} />
      </div>
    </div>
  );
}

NameDateStars.propTypes = {
  username: PropTypes.string.isRequired,
  month: PropTypes.string.isRequired,
  year: PropTypes.string.isRequired,
  stars: PropTypes.number.isRequired,
};

export default NameDateStars;
