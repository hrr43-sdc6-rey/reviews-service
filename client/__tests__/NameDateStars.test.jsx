import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.1';
import NameDateStars from '../components/NameDateStars.jsx';

Enzyme.configure({ adapter: new Adapter() });

describe('NameDateStars', () => {
  it('NameDateStars snapshot should match shallow render', () => {
    const username = 'Bob';
    const month = 'February';
    const year = '2020';
    const stars = 4;
    const wrapper = shallow(
      <NameDateStars
        username={username}
        month={month}
        year={year}
        stars={stars}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
