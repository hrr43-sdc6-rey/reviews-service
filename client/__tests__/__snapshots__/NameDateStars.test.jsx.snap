// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameDateStars NameDateStars snapshot should match shallow render 1`] = `
<div>
  <div
    className="nameDate"
  >
    <p
      className="username"
    >
      Bob
    </p>
    <div
      className="date"
    >
      <div
        className="dot"
      >
        Â·
      </div>
      <div
        className="month"
      >
        February
      </div>
      2020
    </div>
  </div>
  <div
    className="reviewStars"
  >
    <ReviewStars
      stars={4}
    />
  </div>
</div>
`;
